### PR TITLE
🐛(kibana) fix replicas load balancing configuration

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

When using multiple Kibana replicas, targetted replica may fail to find logged user session and returns a HTTP 500 error code.

## Proposal

One must ensure that the server uuid and name are uniques while encryption keys are shared upon replicas.

Source: https://www.elastic.co/guide/en/kibana/current/production.html#load-balancing-kibana

### CODE REVIEW

File: apps/kibana/templates/services/app/deploy.yml.j2 [hunk 1]
```diff
@@ -48,6 +48,10 @@ spec:
         - name: kibana
           image: "{{ kibana_image_name }}:{{ kibana_image_tag }}"
           imagePullPolicy: Always
+          command:
+            - "bash"
+            - "-c"
+            - "SERVER_UUID=$(uuidgen) /usr/local/bin/kibana-docker"
```
Comment (lunika) [inline apps/kibana/templates/services/app/deploy.yml.j2]:
```suggestion
          command: ["/bin/bash"]
          args: ["-c", "SERVER_UUID=$(uuidgen) /usr/local/bin/kibana-docker"]
```

according to the [k8s documentation](https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#run-a-command-in-a-shell), it should be done like this

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,10 @@ Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
 
+### Fixed
+
+- Kibana: fixed configuration to load balance traffic upon replicas
```
Comment (lunika) [inline CHANGELOG.md]:
```suggestion
- Kibana: fix configuration to load balance traffic upon replicas
```
I have a doubt but I think it should be `fix` and not `fixed`.

File: apps/kibana/templates/services/app/configs/kibana.yml.j2 [hunk 3]
```diff
@@ -15,10 +17,13 @@ server.publicBaseUrl: {{ kibana_server_public_base_url | default("https://" + ki
...
-elasticsearch.username: "${KIBANA_ELASTIC_USERNAME}"
-elasticsearch.password: "${KIBANA_ELASTIC_PASSWORD}"
+elasticsearch.username: ${KIBANA_ELASTIC_USERNAME}
+elasticsearch.password: ${KIBANA_ELASTIC_PASSWORD}
 elasticsearch.ssl.certificateAuthorities: ["{{ kibana_ca_path }}"]
-xpack.security.encryptionKey: "${XPACK_SECURITY_ENCRYPTIONKEY}"
+xpack.security.encryptionKey: ${XPACK_SECURITY_ENCRYPTIONKEY}
+xpack.reporting.encryptionKey: ${XPACK_REPORTING_ENCRYPTIONKEY}
+xpack.encryptedSavedObjects.encryptionKey: ${XPACK_ENCRYPTEDSAVEDOBJECTS_ENCRYPTIONKEY}
+xpack.encryptedSavedObjects.keyRotation.decryptionOnlyKeys: ${XPACK_ENCRYPTEDSAVEDOBJECTS_KEYROTATION_DECRYPTIONONLYKEYS}
```
Comment (lunika) [inline apps/kibana/templates/services/app/configs/kibana.yml.j2:26]: `XPACK_ENCRYPTEDSAVEDOBJECTS_KEYROTATION_DECRYPTIONONLYKEYS` is hard to read... it is not possible to split `DECRYPTIONONLYKEYS` ?

File: apps/kibana/templates/services/app/deploy.yml.j2 (hunk 1 above)
Comment (jmaupetit) [reply apps/kibana/templates/services/app/deploy.yml.j2]: Wow, so there is a lot of trays that needs to be fixed!

File: apps/kibana/templates/services/app/configs/kibana.yml.j2 (hunk 3 above)
Comment (jmaupetit) [reply apps/kibana/templates/services/app/configs/kibana.yml.j2:26]: In fact no as there is an automated mapping between environment variables and kibana configuration variables names (underscores became dots).

**Comment by jmaupetit** [comment]
@lunika I've addressed all your comments. Ready for a new review. :pray:

**Review by lunika — APPROVED** [review]

**Pull request merged**